### PR TITLE
chore(*): Bump Spin dependency to v2.4.0

### PR DESCRIPTION
**Pull request by radu-matei**
This commit bumps the Spin dependency to v2.4.0, and addresses the changes in
the trigger trait.

Signed-off-by: Radu Matei <radu@fermyon.com>

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,9 +1,10 @@
-use std::{sync::Arc, collections::HashMap};
+use std::{collections::HashMap, sync::Arc};
```
Comment (radu-matei) [inline src/lib.rs:1]: There are a bunch of formatting changes introduced by running `cargo fmt`.

**Review by itowlson — COMMENTED** [review]
Version bump is fine, but if we're adopting `cargo fmt` on this repo, then I'd prefer to do it as a separate PR as we should also add enforcement in CI etc.  (I'm happy that we do it... just _how_ we do it!)

**Comment by radu-matei** [comment]
Only kept the version bumps in this PR. Thanks!

**Review by itowlson — APPROVED** [review]
Thanks @radu-matei!

**Pull request merged**